PowerEditor/src/Parameters.cpp

```diff
@@ -4511,6 +4511,39 @@ void NppParameters::feedGUIParameters(TiXmlNode *node)
 			if (val)
 				_nppGUI._isMaximized = (lstrcmp(val, TEXT("yes")) == 0);
 		}
+
+		else if (!lstrcmp(nm, TEXT("FindWindowPosition")))
+		{
+			RECT oldRect = _nppGUI._findWindowPos;
+			bool incomplete = true;
+			int i;
+
+			if (element->Attribute(TEXT("left"), &i))
+			{
+				_nppGUI._findWindowPos.left = i;
+
+				if (element->Attribute(TEXT("top"), &i))
+				{
+					_nppGUI._findWindowPos.top = i;
+
+					if (element->Attribute(TEXT("right"), &i))
+					{
+						_nppGUI._findWindowPos.right = i;
+
+						if (element->Attribute(TEXT("bottom"), &i))
+						{
+							_nppGUI._findWindowPos.bottom = i;
+							incomplete = false;
+						}
+					}
+				}
+			}
+			if (incomplete)
+			{
+				_nppGUI._findWindowPos = oldRect;
+			}
+		}

```

Comment:
Perhaps, something like,
```C++
RECT newRect;
if (element->Attribute(TEXT("top"), &newRect.top) &&
    element->Attribute(TEXT("left"), &newRect.left) && 
    element->Attribute(TEXT("right"), &newRect.right) &&
    element->Attribute(TEXT("bottom"), &newRect.bottom))
{
    _nppGUI._findWindowPos = newRect;
}
```

Reply:
I suppose.  My intent was to be very much like the code that reads in the configuration settings for `_appPos` (the main window positioning setting).   If you'll notice, this new code is almost identical to that, except I chose *not* to use the variable name `fuckUp` like Don did (I chose to use the name `incomplete`).  Maybe Don's choice IS better in this case, for descriptiveness, but I don't have the moxie to put something like that into someone else's codebase, even when duplicating code. ;)

Reply:
> except I chose not to use the variable name fuckUp like Don did

Did I? I thought I watch always my language (only C++) :)

However, I prefer @xylographe 's way - more compact and readable.

Reply:
Ok, will change it to be the more compact version.  I think if I would have left the "fuck" variable in there, you would have preferred that version. :)

Reply:
A problem with attempting to change it to the more-compact version is that the RECT struct members are LONG and there is no facility in the xml element Attribute function for reading into that data type.  But, because int and long are the same size (on Windows, and in 32 or 64 bit), one can solve this problem via casting.  But it is ugly now, and I tend to favor the original code in this PR.  What do you guys think about it?:

```
RECT newRect;
if (element->Attribute(TEXT("top"), reinterpret_cast<int *>(&newRect.top)) &&
    element->Attribute(TEXT("left"), reinterpret_cast<int*>(&newRect.left)) &&
    element->Attribute(TEXT("right"), reinterpret_cast<int*>(&newRect.right)) &&
    element->Attribute(TEXT("bottom"), reinterpret_cast<int*>(&newRect.bottom)))
{
    _nppGUI._findWindowPos = newRect;
}

```

Reply:
OK, let's not stick on the small detail - both solutions work for me, so we can keep the way you prefer.

Reply:
@sasumner 
Agreed. First implementation is better than casting-version.